app/scripts/modules/core/src/artifact/ArtifactIconService.ts

```diff
@@ -0,0 +1,29 @@
+const unknownArtifactPath = require('./icons/unknown-type-artifact.svg');
+
+interface IArtifactIcon {
+  type: RegExp;
+  path: string;
+}
+
+export class ArtifactIconService {
+  private static icons: IArtifactIcon[] = [];
+
+  public static registerType(type: RegExp, path: string) {
+    ArtifactIconService.icons.push({ type, path });
+  }
+
+  public static getPath(type: string) {
+    const icon = ArtifactIconService.icons.find(entry => entry.type.test(type));
+    if (icon == null) {
```

Comment:
Minor nit, but `find` returns `undefined` when it doesn't find anything.  This will of course still work given the comparison is `==`.